api-contract/contract.yaml

```diff
@@ -348,7 +352,167 @@ paths:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Friend'
+        required: true
+      security:
+        - bearer_auth: []
+      responses:
+        default:
+          description: successful operation
+          content: {}
+    delete:
+      tags:
+      - friend
+      summary: Delete a friend
+      description: This can be done by the logged in user.
+      operationId: deleteFriend
+      requestBody:
+        description: Delete friend object
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Friend'
+        required: true
+      security:
+        - bearer_auth: []
+      responses:
+        default:
+          description: successful operation
+          content: {}
+    put:
+      tags:
+      - friend
+      summary: Block a friend
+      description: This can be done by the logged in user.
+      operationId: blockFriend
+      requestBody:
+        description: Block friend object
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Friend'
+        required: true
+      security:
+        - bearer_auth: []
+      responses:
+        default:
+          description: successful operation
+          content: {}
+  /friend/all:
+    get:
+      tags:
+      - friend
+      summary: View a list of friends
+      description: This can be done by the logged in user.
+      operationId: viewFriend
+      requestBody:
+        description: View friend object
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/Friend'
+        required: true
+      security:
+        - bearer_auth: []
+      responses:
+        default:
+          description: successful operation
+          content: {}
+  /group:
+    post:
+      tags:
+      - group
+      summary: Create a group of friends
+      description: This can be done by the logged in user.
+      operationId: createGroup
+      requestBody:
+        description: Create group object
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Group'
+        required: true
+      security:
+        - bearer_auth: []
+      responses:
+        default:
+          description: successful operation
+          content: {}
+    put:
+      tags:
+      - group
+      summary: Edit a group of friends
+      description: This can be done by the logged in user.
+      operationId: editGroup
+      requestBody:
+        description: Edit group object
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Group'
+        required: true
+      security:
+        - bearer_auth: []
+      responses:
+        default:
+          description: successful operation
+          content: {}
```

Comment:
When we update a group, we need two things:
* the groupId: this can be a param (missing)
* the group schema: just like you have it now (no changes in this part)